src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/HttpUtilities.cs

```diff
@@ -22,7 +22,7 @@ internal static class HttpUtilities
             string.Equals(scheme, "wss", StringComparison.OrdinalIgnoreCase);
 
         internal static bool IsSupportedProxyScheme(string scheme) =>
-            string.Equals(scheme, "http", StringComparison.OrdinalIgnoreCase) || IsSocksScheme(scheme);
+            string.Equals(scheme, "http", StringComparison.OrdinalIgnoreCase) || string.Equals(scheme, "https", StringComparison.OrdinalIgnoreCase) || IsSocksScheme(scheme);
```

Comment:
Likely worth updating the exception message as well
https://github.com/dotnet/runtime/blob/c8efa1204de234d8780ae993612b5f005e397651/src/libraries/System.Net.Http/src/Resources/Strings.resx#L372-L373